functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionOnMultipleNodesSpec.groovy

```diff
@@ -0,0 +1,191 @@
+                 "jobUuidOption"     : "preserve"])
+        //wait for ansible node to be available
+        waitingResourceEnabled(TEST_PROJECT, "ssh-node")
+    }
+
+    def "Create a job that executes on all nodes"() {
+        given:
+        def jobName = UUID.randomUUID().toString()
+        def jobXml = """
+        <joblist>
+           <job>
+              <name>${jobName}</name>
+              <group>api-test</group>
+              <description></description>
+              <loglevel>INFO</loglevel>
+              <multipleExecutions>true</multipleExecutions>
+              <dispatch>
+                <threadcount>1</threadcount>
+                <keepgoing>true</keepgoing>
+              </dispatch>
+              <nodefilters>
+                    <filter>.*</filter>
+               </nodefilters>
+              <executionEnabled>true</executionEnabled>
+              <sequence>
+                <command>
+                  <exec>echo 0</exec>
+                </command>
+              </sequence>
+           </job>
+        </joblist>"""
+
+        when:
+        def response = JobUtils.createJob(TEST_PROJECT, jobXml, client)
+
+        then:
+        response.successful
+        def jobId = MAPPER.readValue(response.body().string(), CreateJobResponse.class).getSucceeded().get(0).id
+        def completedJob = runJobAndWait(jobId)
+        // Ensure it was ran on the local node and two nodes added by the setup
+        getOrderedNodesListExecutedOn(completedJob).size() == 3
+    }
+
+    def "Create a job that executes on nodes that were not excluded"() {
+        given:
+        def jobName = UUID.randomUUID().toString()
+        def jobXml = """
+        <joblist>
+           <job>
+              <name>${jobName}</name>
+              <group>api-test</group>
+              <description></description>
+              <loglevel>INFO</loglevel>
+              <multipleExecutions>true</multipleExecutions>
+              <dispatch>
+                <threadcount>1</threadcount>
+                <keepgoing>true</keepgoing>
+              </dispatch>
+               <nodefilters>
+                   <filter>.*</filter>
+                   <filterExclude>tags: "executor-test"</filterExclude>
+               </nodefilters>
+              <executionEnabled>true</executionEnabled>
+              <sequence>
+                <command>
+                  <exec>echo 0</exec>
+                </command>
+              </sequence>
+           </job>
+        </joblist>"""
+
+        when:
+        def response = JobUtils.createJob(TEST_PROJECT, jobXml, client)
+
+        then:
+        response.successful
+        def jobId = MAPPER.readValue(response.body().string(), CreateJobResponse.class).getSucceeded().get(0).id
+        def completedJob = runJobAndWait(jobId)
+        // Should run the local node only since the filter excluded the nodes added by the setup
+        getOrderedNodesListExecutedOn(completedJob).size() == 1
```

Comment:
same suggestion as the previous test

Reply:
The name of the local node is dynamic, thus I cannot do a simple list comparison.